### PR TITLE
Add Makefile modified from Cosmos-SDK

### PR DESCRIPTION
**Pull request by jackzampolin**
This PR adds a makefile with some standard targets to build the project. Its copy-pasta (minus a bunch) from cosmos-sdk. Also the testing make targets are all commented out. I thought this could be a starting point and we could build out from here. I'm also happy to rethink the whole thing too.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,153 @@
+PACKAGES=$(shell go list ./... | grep -v '/vendor/')
+PACKAGES_NOCLITEST=$(shell go list ./... | grep -v '/vendor/' | grep -v github.com/cosmos/ethermint/cmd/gaia/cli_test)
```
Comment (alexanderbez) [inline Makefile]: We can remove this line...for now.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+UNCONVERT_CHECK := $(shell command -v unconvert 2> /dev/null)
+INEFFASSIGN_CHECK := $(shell command -v ineffassign 2> /dev/null)
+MISSPELL_CHECK := $(shell command -v misspell 2> /dev/null)
+ERRCHECK_CHECK := $(shell command -v errcheck 2> /dev/null)
+UNPARAM_CHECK := $(shell command -v unparam 2> /dev/null)
+GOCYCLO_CHECK := $(shell command -v gocyclo 2> /dev/null)
+
+all: get-tools get-vendor-deps install
+
+########################################
```
Comment (alexanderbez) [inline Makefile]: I think we can remove the comment-y blocks for now 👍

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+
+########################################
+### Documentation
+
+godocs:
+	@echo "--> Wait a few seconds and visit http://localhost:6060/pkg/github.com/cosmos/ethermint"
+	godoc -http=:6060
+
+########################################
```
Comment (alexanderbez) [inline Makefile]: Let's remove the testing stuff for now until we need it 👍

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+build:
+ifeq ($(OS),Windows_NT)
+	go build $(BUILD_FLAGS) -o build/ethermint.exe ./*.go
+else
+	go build $(BUILD_FLAGS) -o build/ethermint ./*.go
+endif
+
+install:
+	go install $(BUILD_FLAGS) ./*.go
+
```
Comment (alexanderbez) [inline Makefile:38]: Can we also add a `clean` target, that say, removes `./build` and `./vendor`?

**Review by alexanderbez — CHANGES_REQUESTED** [review]
Thanks a lot for this @jackzampolin, left a few minor requests 👍

File: Makefile (hunk 4 above)
Comment (jackzampolin) [reply Makefile:38]: Yup!

File: Makefile (hunk 3 above)
Comment (jackzampolin) [reply Makefile]: Good call.

File: Makefile (hunk 2 above)
Comment (jackzampolin) [reply Makefile]: Going to clean up all the comments

File: Makefile (hunk 1 above)
Comment (jackzampolin) [reply Makefile]: Yup!

**Comment by jackzampolin** [comment]
Addressed comments!

**Review by alexanderbez — APPROVED** [review]
LGTM 👍

**Pull request merged**